roles/reproducer/tasks/configure_crc.yml

```diff
@@ -5,108 +5,62 @@
-      become: true
-      vars:
-        _config_file: >-
-          {{
-            _dnsmasq.stat.exists |
-            ternary(_dnsmasq.stat.path, '/etc/dnsmasq.d/crc-dnsmasq.conf')
-          }}
-      block:
-        - name: Configure dns forwarders
-          become: true
-          ansible.builtin.blockinfile:
-            path: "{{ _config_file }}"
-            block: |-
-              {% if cifmw_reproducer_dns_servers %}
-              {% for dns_server in cifmw_reproducer_dns_servers %}
-              server={{ dns_server }}
-              {% endfor %}
-              {% endif %}
-
-        - name: Configure local DNS for CRC pod
-          register: last_modification
-          ansible.builtin.replace:
-            path: "{{ _config_file }}"
-            regexp: "192.168.130.11"
-            replace: "{{ _crc.ip_v4 }}"
-
-        - name: Ensure dnsmasq listens on correct interfaces
-          ansible.builtin.replace:
-            path: "{{ _config_file }}"
-            regexp: "listen-address={{ _crc.ip_v4 }}"
-            replace: "listen-address={{ _crc.ip_v4 }},127.0.0.1"
-
-    - name: Reboot CRC node
-      become: true
-      ansible.builtin.reboot:
+- name: Remove CRC managed zone delegation
+  become: true
+  notify: Restart NetworkManager
+  ansible.builtin.file:
+    path: ""
+    state: absent
 
-- name: Ensure hypervisor has the right CRC IP
+- name: Remove entry from /etc/hosts
   become: true
   vars:
     _parsed: "{{ _net_conf.content | b64decode | from_yaml}}"
     _crc: "{{ _parsed.instances['crc-0'].networks.ctlplane }}"
   ansible.builtin.blockinfile:
+    state: absent
     path: "/etc/hosts"
     marker: "# {mark}"
     marker_begin: "Added by CRC"
     marker_end: "End of CRC section"
-    block: >-
-      {{ _crc.ip_v4 }} api.crc.testing
-      canary-openshift-ingress-canary.apps-crc.testing
-      console-openshift-console.apps-crc.testing
-      default-route-openshift-image-registry.apps-crc.testing
-      downloads-openshift-console.apps-crc.testing
-      oauth-openshift.apps-crc.testing
+
+- name: Inject our DNS in cifmw-dnsmasq
+  vars:
+    _parsed: "{{ _net_conf.content | b64decode | from_yaml}}"
+    _crc: "{{ _parsed.instances['crc-0'].networks.ctlplane }}"
+    _act: 'install'
+  block:
+    - name: Inject DNS records for crc
+      when:
+        - ip | length > 0
+      vars:
+        cifmw_dnsmasq_address:
+          - domains:
+              - api.crc.testing
+              - api-int.crc.testing
+              - canary-openshift-ingress-canary.apps-crc.testing
+              - console-openshift-console.apps-crc.testing
+              - default-route-openshift-image-registry.apps-crc.testing
+              - downloads-openshift-console.apps-crc.testing
+              - oauth-openshift.apps-crc.testing
+            ipaddr: "{{ ip }}"
+            state: present
+      ansible.builtin.include_role:
+        name: "dnsmasq"
+        tasks_from: "manage_address.yml"
```

Comment:
I don't this all of these should be `address`
--address is `Specify an IP address to return for any host in the given domains.`

We should use --host-record for api.crc.testing and api-int.crc.testing ?

And then one address entry with `apps-crc.testing` should be enough?

Reply:
hm, maybe `address` for `crc.testing` then? Not 100% sure about the used domains/others. We indeed have "some" hosts under `apps-crc.testing` - so that's `address` for sure.

~> maybe using address to cover the whole `crc.testing` would be a thing?

Reply:
updated - we have "just" 2 addresses now (and yeah, I totally mixed things in the previous iteration). Lemme know if that's OK now.

Reply:
We should be careful with the wildcard - I think only apps-crc.testing should be wildcard.

See: https://github.com/openstack-k8s-operators/install_yamls/commit/cd1c8e79828b49eb85234490e754cea1cd1e8590

I used a wildcard - `address=/sno.lab.example.com/${SNO_HOST_IP}` - this cause issues with resolving internally in OCP. Hence I had to change to to only wildcard `apps.$DOMAIN`.

Reply:
ok, will check that out, but then we'll want to get your dual-stack patch in first I think.